### PR TITLE
TF examples and bug fixes

### PR DESCRIPTION
**Pull request by gnavvy**
This PR includes several fixes for the VAO, TransformFeedback and Program APIs in luma.gl, together with a minimal working example in the examples/transform-feedback directory.

Per discussion, there will be a few more APIs added to allow user-defined location -> buffer mappings, but want to have the bugs fixed at least before the 4.0.0 release.

### CODE REVIEW

File: docs/api-reference/webgl/transform-feedback.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ const program = new Program(gl, {
   varyings: ['gl_Position'],
 });
 
+[//]: # transformFeedback is not defined yet
```
Comment (shaojingli) [inline docs/api-reference/webgl/transform-feedback.md]: Is this line necessary?

Comment (gnavvy) [inline docs/api-reference/webgl/transform-feedback.md]: no, it was a note that we did not declare transformFeedback before using it in the doc.
will remove for this PR.

**Comment by ibgreen** [comment]
@gnavvy - tried to work in this branch but had serious problems rebasing it to latest master. Copied your example and finally got it working. Recommend you cut a new branch from master to continue.

To run:
``
cd demo
npm start
```
Go to `http://localhost:3000/#/examples/core-examples/tranform-feedback`

**Comment by gnavvy** [comment]
> Recommend you cut a new branch from master to continue.
will do.

**Pull request closed without merging**